Tools/log_encryption/download_logs.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import time
+import shutil
+import threading
+from pymavlink import mavutil
+from argparse import ArgumentParser
+
+class MavlinkLogDownloader:
+    def __init__(self, connection_url, output_dir):
+        self.connection_url = connection_url
+        self.output_dir = output_dir
+        self.encrypted_dir = os.path.join(output_dir, "encrypted")
+        self.running = True
+
+        # Ensure directories exist
+        os.makedirs(self.output_dir, exist_ok=True)
+        os.makedirs(self.encrypted_dir, exist_ok=True)
+
+        # Handle serial or UDP connections
+        if connection_url.startswith("serial:"):
+            parts = connection_url.split(":")
+            if len(parts) != 3:
+                raise ValueError("Invalid serial connection URL. Use format: serial:/dev/ttyACM1:57600")
+            device, baudrate = parts[1], int(parts[2])
+            self.mav = mavutil.mavlink_connection(device, baud=baudrate)
```

Comment:
I think mavutil.mavlink_connection already hadnles the serial connection string format, ie `serial:/dev/ttyACM1:57600` so the split on `:` might be unnecessary. Also we should use an arg to specify the source_system or maybe change the default to 254 since QGC uses 255 and this would be buggy if used at the same time as QGC.